ops/pebble.py

```diff
@@ -43,22 +43,243 @@
 import urllib.parse
 import urllib.request
 import warnings
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    BinaryIO,
+    Callable,
+    Dict,
+    Generator,
+    Iterable,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    TextIO,
+    Tuple,
+    TypeVar,
+    Union,
+)
 
 from ops._private import yaml
 from ops._vendor import websocket
 
+if TYPE_CHECKING:
+    _Boundary = Tuple[int, int, bool]
+    from email.message import Message
+
+    from typing_extensions import Literal, Protocol, TypedDict
+
+    # callback types for _MultiParser header and body handlers
+    class _BodyHandler(Protocol):
+        def __call__(self, data: bytes, done: bool = False) -> None: ...  # noqa
+
+    _HeaderHandler = Callable[[bytes], None]
+    _StrOrByt = Union[str, bytes]
```

Comment:
Isn't this `typing.AnyStr`?

Reply:
Well type vars can only be used as variables, e.g.

```
def foo(s: AnyStr) -> AnyStr: ...
```

but if you do:
```
a = ''  # type: AnyStr
```

pyright will say:
![image](https://user-images.githubusercontent.com/6230162/173299474-a8fb4f20-20b5-4273-97f1-e94cea19b695.png)

see discussion here for some context
https://gitter.im/python/typing?at=5d02668a72e573260795a800